sysvars/sysvars_ec2.go

```diff
@@ -15,43 +15,51 @@
 package sysvars
 
 import (
+	"context"
 	"fmt"
-	"net/http"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/ec2metadata"
-	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
 	"github.com/cloudprober/cloudprober/logger"
 )
 
+// loadAWSConfig will evaluate if we should apply retries or not to the AWS config
+func loadAWSConfig(ctx context.Context, tryHard bool) (aws.Config, error) {
+	retryMaxAttempts := 3
+
+	// if we are not trying hard, then we can disable any retries
+	if !tryHard {
+		retryMaxAttempts = 0
+	}
+
+	return config.LoadDefaultConfig(ctx, config.WithRetryMaxAttempts(retryMaxAttempts))
+}
+
 var ec2Vars = func(sysVars map[string]string, tryHard bool, l *logger.Logger) (bool, error) {
-	cfg := &aws.Config{}
-	// If not trying hard (cloud_metadata != ec2), use shorter timeout.
+	ctx := context.Background()
+
+	// if we are not trying hard, then we can apply a strict timeout on the AWS API calls
```

Comment:
Previous comment probably added more color to try hard :)
```suggestion
	// If not trying hard (cloud_metadata != ec2), use shorter timeout.
```

Reply:
your right, honestly, I overlooked the previous comments and forgot them - my oversight